mitosheet/src/mito/components/taskpanes/Merge/MergeTaskpane.tsx

```diff
@@ -106,15 +98,103 @@ export const getDefaultMergeParams = (sheetDataArray: SheetData[], _sheetIndexOn
     }
 }
 
+const getValidMergeParams = (params: MergeParams, sheetDataArray: SheetData[]): MergeParams => {
+    const {
+        merge_key_column_ids,
+        selected_column_ids_one,
+        selected_column_ids_two 
+    } = params;
 
-const MergeTaskpane = (props: MergeTaskpaneProps): JSX.Element => {
+    const sheetDataOne = sheetDataArray[params.sheet_index_one];
+    const sheetDataTwo = sheetDataArray[params.sheet_index_two];
+
+    const validMergeKeyColumnIDs = merge_key_column_ids.filter(([mergeKeyColumnIDOne, mergeKeyColumnIDTwo]) => {
+        return sheetDataOne.columnIDsMap[mergeKeyColumnIDOne] !== undefined && sheetDataTwo.columnIDsMap[mergeKeyColumnIDTwo] !== undefined
+    });
+
+    const validSelectedColumnIDsOne = selected_column_ids_one.filter((columnID) => {
+        return sheetDataOne.columnIDsMap[columnID] !== undefined
+    });
+
+    const validSelectedColumnIDsTwo = selected_column_ids_two.filter((columnID) => {
+        return sheetDataTwo.columnIDsMap[columnID] !== undefined
+    });
+
+    return {
+        ...params,
+        merge_key_column_ids: validMergeKeyColumnIDs,
+        selected_column_ids_one: validSelectedColumnIDsOne,
+        selected_column_ids_two: validSelectedColumnIDsTwo
+    }
+}
+
+const getEditMergeWarnings = (params: MergeParams, sheetDataOne: SheetData, sheetDataTwo: SheetData): {
+    mergeKeyWarnings?: string[],
+    selectedColumnsOneWarnings?: string,
+    selectedColumnsTwoWarnings?: string,
+} => {
+    const {
+        merge_key_column_ids,
+        selected_column_ids_one,
+        selected_column_ids_two 
+    } = params;
+
+    const invalidMergeKeys = merge_key_column_ids.filter(([mergeKeyColumnIDOne, mergeKeyColumnIDTwo]) => {
+        return sheetDataOne.columnIDsMap[mergeKeyColumnIDOne] === undefined || sheetDataTwo.columnIDsMap[mergeKeyColumnIDTwo] === undefined
+    });
+    const mergeKeyWarnings = invalidMergeKeys.map(([mergeKeyColumnIDOne, mergeKeyColumnIDTwo]) => {
+        const sheetOneColumnHeader = sheetDataOne.columnIDsMap[mergeKeyColumnIDOne];
+        const sheetTwoColumnHeader = sheetDataTwo.columnIDsMap[mergeKeyColumnIDTwo];
+        if (sheetOneColumnHeader === undefined && sheetTwoColumnHeader === undefined) {
+            return `The merge keys ${mergeKeyColumnIDOne} and “${mergeKeyColumnIDTwo}” were removed because they no longer exist in “${sheetDataOne.dfName}” and “${sheetDataTwo.dfName}”.`
```

Comment:
I think (columnHeaderOne, columnHeaderTwo) also works here

Reply:
I wanted to do that too but the if statement above is checking that they're `undefined`!